src/D2L.CodeStyle.Analyzers/Pinning/PinnedAnalyzerHelper.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+
+namespace D2L.CodeStyle.Analyzers.Pinning {
+	public static class PinnedAnalyzerHelper {
+
+		public const string PinnedAttributeName = "D2L.CodeStyle.Annotations.Pinning.PinnedAttribute";
+
+		public static bool TryGetPinnedAttribute( ISymbol classSymbol, INamedTypeSymbol pinnedAttributeSymbol, out AttributeData? attribute ) {
+			attribute = null;

```

Comment:
nit: prefer setting this value just before returning `false`. Setting it here is both less performant (unused assignment on `true` path), and can mask logic errors where the value otherwise should have been set to something different.

Reply:
alternatively, you can change the definition to `AttributeData? GetPinnedAttribute(...)`. I'm usually a fan of the `TryABC()` pattern, but with the C# nullablity types it's a bit odd because returning `false` already tells you that, so the caller has to check both the return type and the `out` parameter (or use `attribute!.something()` everywhere)